### PR TITLE
Outdated

### LINKED ISSUE #478: Spotguide Catalog support
- [X] Provide Catalogs List endpoint
- [X] Provide Catalogs Details endpoint
- [X] Configurable catalogs repository

### PR DESCRIPTION
**Pull request by tarokkk**
- [x]  GET /orgs/id/catalogs endpoint
- [x]  GET /orgs/{id}/catalogs/{name} endpoint
- [x] Initialise catalog repo on first query in organisation scope
- [x] spotguide.yaml/spotguide.json available
  - Resources
  - Depends
  - Options

fixes #478

### CODE REVIEW

**Comment by baluchicken** [comment]
Please update Postman collection as well.

File: api/functions.go [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package api
+
+import (
+	"github.com/banzaicloud/pipeline/cluster"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+// Apply the given functions to the cluster
+func ApplyFunctions(c *gin.Context) {
```
Comment (baluchicken) [inline api/functions.go]: Wrong doc it should start with the function name

File: api/cluster.go [hunk 2]
```diff
@@ -98,8 +100,7 @@ func GetCommonClusterNameFromRequest(c *gin.Context) (string, bool) {
 	return clusterName, true
 }
 
-// CreateCluster creates a K8S cluster in the cloud
-func CreateCluster(c *gin.Context) {
+func CreateClusterRequest(c *gin.Context) {
```
Comment (baluchicken) [inline api/cluster.go]: Doc missing

File: api/catalog.go [hunk 3]
```diff
@@ -0,0 +1,59 @@
+package api
```
Comment (baluchicken) [inline api/catalog.go]: Doc missing, circle ci will fail because of this

File: api/applications.go [hunk 4]
```diff
@@ -0,0 +1,131 @@
+package api
```
Comment (baluchicken) [inline api/applications.go]: Doc is missing

File: api/applications.go [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+	}
+	c.JSON(http.StatusOK, response)
+	return
+}
+
+func ApplicationDetails() {
+
+}
+
+//Create Application
```
Comment (baluchicken) [inline api/applications.go]: Wrong doc

File: application/applications.go [hunk 6]
```diff
@@ -0,0 +1,226 @@
+package application
+
```
Comment (baluchicken) [inline application/applications.go]: Doc is missing

File: catalog/catalog.go [hunk 7]
```diff
@@ -0,0 +1,266 @@
...
+	"regexp"
+	"strings"
+)
+
+const CatalogRepository = "catalog"
+const CatalogRepositoryUrl = "http://kubernetes-charts.banzaicloud.com/branch/spotguide"
+
+var CatalogPath = "./" + CatalogRepository
+
+//TODO when the API fixed this needs to move to banzai-types
```
Comment (baluchicken) [inline catalog/catalog.go]: Is this fixed?

Comment (tarokkk) [reply catalog/catalog.go]: It won't until the UI is fully functional

**Comment by tarokkk** [comment]
Due to many conflicts new PR will be opened

**Pull request closed without merging**